src/main/java/net/mcreator/workspace/WorkspaceFolderManager.java

```diff
@@ -82,56 +92,20 @@ public File getOtherTextureFile(String textureIdentifier) {
 	}
 
 	public File[] getArmorTextureFilesForName(String armorTextureName) {
-		return new File[] { new File(getArmorTexturesDir(), armorTextureName + "_layer_1.png"),
-				new File(getArmorTexturesDir(), armorTextureName + "_layer_2.png") };
-	}
-
-	public List<File> getBlockTexturesList() {
-		return listPNGsInDir(getBlocksTexturesDir());
-	}
-
-	public List<File> getItemTexturesList() {
-		return listPNGsInDir(getItemsTexturesDir());
-	}
-
-	public List<File> getArmorTexturesList() {
-		return listPNGsInDir(getArmorTexturesDir());
+		return new File[] { new File(getTexturesDirTypeFromSectionType(TextureSection.ARMOR), armorTextureName + "_layer_1.png"),
+				new File(getTexturesDirTypeFromSectionType(TextureSection.ARMOR), armorTextureName + "_layer_2.png") };
 	}
 
-	public List<File> getOtherTexturesList() {
-		return listPNGsInDir(getOtherTexturesDir());
+	public List<File> getTexturesListFromSectionType(TextureSection section) {
+		return listPNGsInDir(getTexturesDirTypeFromSectionType(section));
 	}
 
 	public void removeStructure(String name) {
 		new File(getStructuresDir(), name + ".nbt").delete();
 	}
 
-	private static List<File> listPNGsInDir(@Nullable File dir) {
-		if (dir == null)
-			return Collections.emptyList();
-
-		List<File> retval = new ArrayList<>();
-		File[] block = dir.listFiles();
-		for (File element : block != null ? block : new File[0])
-			if (element.getName().endsWith(".png"))
-				retval.add(element);
-		return retval;
-	}
-
-	@Nullable public File getBlocksTexturesDir() {
-		return GeneratorUtils.getSpecificRoot(workspace, workspace.getGeneratorConfiguration(), "block_textures_dir");
-	}
-
-	@Nullable public File getItemsTexturesDir() {
-		return GeneratorUtils.getSpecificRoot(workspace, workspace.getGeneratorConfiguration(), "item_textures_dir");
-	}
-
-	@Nullable public File getArmorTexturesDir() {
-		return GeneratorUtils.getSpecificRoot(workspace, workspace.getGeneratorConfiguration(), "armor_textures_dir");
-	}
-
-	@Nullable public File getOtherTexturesDir() {
-		return GeneratorUtils.getSpecificRoot(workspace, workspace.getGeneratorConfiguration(), "other_textures_dir");
+	@Nullable public File getTexturesDirTypeFromSectionType(TextureSection section) {

```

Comment:
```suggestion
	@Nullable public File getTexturesFolderForSectionType(TextureSection section) {
```

Reply:
You didn't have to merge my suggestion, I just gave an idea. Now you forgot to rename this everywhere it's used.